apps/ai-image-generator/backend/scripts/validate-actions.js

```diff
@@ -0,0 +1,33 @@
+const manifest = require('../../contentful-app-manifest.json');
+
+const validateActions = () => {
+  const requiredProperties = ['id', 'path'];
+  const uniqueValues = new Set();
+
+  manifest.actions.forEach((action) => {
+    requiredProperties.forEach((property) => {
+      if (!action.hasOwnProperty(property)) {
+        throw new Error(`Action with name: '${action.name}' is missing the '${property}' property`);
+      }
+    });
+
+    const { id, path } = action;
+
+    if (uniqueValues.has(id)) {
+      throw new Error(`Duplicate action id: '${id}'`);
+    }
+    if (uniqueValues.has(path)) {
+      throw new Error(`Duplicate action path: '${path}'`);
+    }
+
+    uniqueValues.add(path);
+    uniqueValues.add(id);
+  });
+};
+
+const main = async () => {
+  console.log('Building app actions');
+  validateActions();
```

Comment:
Also would love to know more about why we don't care to look for esBuild configs

Reply:
Yes! We ditch esbuild entirely and just use `tsc` (same way most other typescript projects are built today. We could use esbuild if we wanted or any build. But it's separate from this script now. We just build in our project.